modules/public_ip/outputs.tf

```diff
@@ -0,0 +1,9 @@
+output "pip_ids" {
+  description = "The identifiers of the created or sourced Public IP Addresses."
+  value       = { for k, v in local.public_ips : k => v.id }
+}
```

Comment:
Maybe for public IP it could valuable to output also [ip_address](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/public_ip#ip_address) ?

Reply:
thank you @acelebanski